lib/src/git.rs

```diff
@@ -1299,6 +1303,80 @@ fn build_index_from_merged_tree(
     Ok(index)
 }
 
+/// Convenience wrapper to edit Git configuration files.
+///
+/// Note that the resulting configuration changes are *not* persisted to the
+/// originating [`gix::Repository`]! The repository must be reloaded with the
+/// new configuration. Functions using this API should ideally take a
+/// [`gix::Repository`] by value to hint this to the caller.
+pub struct GitConfigEditor {
+    config: gix::config::File<'static>,
+    config_meta: gix::config::file::Metadata,
+    repo_path: PathBuf,
+}
+
+impl GitConfigEditor {
+    pub fn new(git_repo: &gix::Repository) -> Self {
+        let repo_path = git_repo.path().to_owned();
+        let config = git_repo.config_snapshot().clone();
+        let config_meta = Arc::unwrap_or_clone(config.meta_owned());
+        Self {
+            config,
+            config_meta,
+            repo_path,
+        }
+    }
+
+    pub fn remove_remote_sections(&mut self, remote_name: &str) {
+        let Some(iter) = self.sections_and_ids_by_name("remote") else {
+            return;
+        };
+        // Note: This removes entire sections rather than just the basic Git options, so
+        // this can clobber user data. Ideally `gix` would expose the more complex logic
+        // it uses for changing existing remotes.
+        let section_ids_to_remove: Vec<_> = iter
+            .filter(|(section, _)| {
+                section.header().subsection_name() == Some(BStr::new(remote_name))
+            })
+            .map(|(_, id)| id)
+            .collect();
+        for id in section_ids_to_remove {
+            self.remove_section_by_id(id)
+                .expect("removed section to exist");
+        }
+    }
+
+    pub fn save(self) -> std::io::Result<()> {
+        let mut temp_config_file = NamedTempFile::with_prefix_in("config", self.repo_path)?;
+        self.config
+            .write_to_filter(&mut temp_config_file, |section| {
+                section.meta() == &self.config_meta
+            })?;
+        temp_config_file
+            .persist(
+                self.config_meta
+                    .path
+                    .expect("Git repository to have a config file"),
+            )
+            .map_err(|err| err.error)?;
+        Ok(())
+    }
+}
+
+impl Deref for GitConfigEditor {
```

Comment:
nit: suppose `GitConfigEditor` is mainly designed for local use, the API ergonomics provided by `Deref` wouldn't be needed?

Reply:
I found that it helped a bit with the functions that were already quite noisy from the error handling. My expectation was that we’d end up having cause to modify the Git configuration more in future and so a nice API would be helpful, but I don’t know how likely that is. I can drop it if it’s unwanted, although now the type is just a wrapper around `gix::config::File` (so it could maybe even be an extension trait instead or something).

Reply:
or a couple of free functions? I don't feel strongly, but I tend to avoid using `Deref` unless the type is like smart pointers. Hopefully there will be a higher-level abstraction in gix in future.

Reply:
Yeah, I’ve replaced it with a couple free functions, since it kinda ended up withering to nothing. It used to be more elaborate.